pandas/tests/tools/test_to_datetime.py

```diff
@@ -125,24 +125,21 @@ def test_to_datetime_format_YYYYMMDD_with_nat(self, cache):
         expected[2] = np.nan
         ser[2] = np.nan
 
-        result = to_datetime(ser, format="%Y%m%d", cache=cache)
-        tm.assert_series_equal(result, expected)
+        with pytest.raises(ValueError, match=None):
+            to_datetime(ser, format="%Y%m%d", cache=cache)
 
         # string with NaT
         ser2 = ser.apply(str)
         ser2[2] = "nat"
-        result = to_datetime(ser2, format="%Y%m%d", cache=cache)
-        tm.assert_series_equal(result, expected)
+        with pytest.raises(ValueError, match=None):
+            to_datetime(ser2, format="%Y%m%d", cache=cache)
 
     def test_to_datetime_format_YYYYMMDD_ignore(self, cache):
         # coercion
         # GH 7930
         ser = Series([20121231, 20141231, 99991231])
+        expected = Series([20121231, 20141231, 99991231], dtype=object)
         result = to_datetime(ser, format="%Y%m%d", errors="ignore", cache=cache)
-        expected = Series(
-            [datetime(2012, 12, 31), datetime(2014, 12, 31), datetime(9999, 12, 31)],
-            dtype=object,
-        )
         tm.assert_series_equal(result, expected)
```

Comment:
this didn't look right to begin with, if the input is invalid, errors='ignore' should return the input

that's what we do for literally any other format, e.g.
```
In [46]: to_datetime(Series([201212, 201412, 999912]), errors='ignore', format='%Y%m')
Out[46]:
0    201212
1    201412
2    999912
dtype: object
```